### PR TITLE
Updated docs for refresh block

### PR DESCRIPTION
**Pull request by bennettgo**
fixes https://github.com/workato/issues/issues/11982

### CODE REVIEW

File: docs/developing-connectors/sdk/authentication/oauth2-authentication.md [hunk 1]
```diff
@@ -341,38 +341,49 @@ In the below example, the Namely API asks for the `refresh_token` to be appended
...
+                        refresh_token: refresh_token,
+                        redirect_uri: "https://www.workato.com/oauth/callback"
+                      ).
+                      request_format_www_form_urlencoded
+        [
+          {
+            access_token: response["access_token"],
+            refresh_token: response["refresh_token"]
+          },
+          { instance_id: nil }
```
Comment (eeshansim) [inline docs/developing-connectors/sdk/authentication/oauth2-authentication.md:380]: better to leave a comment
```ruby
# Optional. Will be merged into connection hash.
```

File: docs/developing-connectors/sdk/authentication/oauth2-authentication.md [hunk 2]
```diff
@@ -341,38 +341,49 @@ In the below example, the Namely API asks for the `refresh_token` to be appended
...
+                      code: auth_code
+                    ).
+                    request_format_www_form_urlencoded
+
+        [
+          {
+            access_token: response["access_token"],
+            refresh_token: response["refresh_token"]
+          },
+          nil,
```
Comment (eeshansim) [inline docs/developing-connectors/sdk/authentication/oauth2-authentication.md:358]: hint for what each of these represent resource_owner_id and optional hash to be merged into connection.

File: docs/developing-connectors/sdk/authentication/oauth2-authentication.md [hunk 3]
```diff
@@ -341,38 +341,49 @@ In the below example, the Namely API asks for the `refresh_token` to be appended
...
+        response = post("https://#{connection["domain"]}.namely.com/api/v1/oauth2/token").
+                      payload(
+                        grant_type: "refresh_token",
+                        client_id: connection["client_id"],
+                        client_secret: connection["client_secret"],
+                        refresh_token: refresh_token,
+                        redirect_uri: "https://www.workato.com/oauth/callback"
+                      ).
+                      request_format_www_form_urlencoded
+        [
```
Comment (eeshansim) [inline docs/developing-connectors/sdk/authentication/oauth2-authentication.md:375]: Comment on backward compatibility. What happens to existing refresh code blocks that do not return an array.

Comment (bennettgo) [reply docs/developing-connectors/sdk/authentication/oauth2-authentication.md:375]: addressed in the section below

**Pull request merged**